### ISSUE TITLE
libsodium error

### ISSUE THREAD

**Issue by hadaroren11** [keystone-enclave/keystone-demo#8]
hey,
i’m getting the following error when i compile (make command in quickstart.sh)
![image](https://user-images.githubusercontent.com/57223550/70993469-c5171000-20d4-11ea-88b4-ed8a4ff3b8b7.png)
you got any solution or idea why i get this error?

**Comment by dkohlbre** [comment]
The most likely reason for that error is that the wrong compiler was used while building the client libsodium (x86_64 vs RISC-V).
Can you give the specific checkouts of Keystone and Keystone Demo that you are on?
And did you run anything in the demo directory besides running `quickstart.sh` one time?

**Comment by hadaroren11** [comment]
in Keystone i’m in : commit ab4e7580c541a7c9798648c5423b6729c82a4c21 (branch 'origin/dev’)
in keystone-demo i’m in : commit 640098894aa644fdf583a37bd84e999baf62999a (branch 'origin/master’).
i ran only quickstart script...
should i change something about the compiler? if the answer is yes so you know what?

**Comment by dkohlbre** [comment]
I was able to reproduce this only by running `quick-start.sh` multiple times.
Make sure you have a clean checkout of keystone-demo, export `KEYSTONE_DIR`, and then run `quick-start.sh` one time exactly.

You will, currently, run into another issue after that due to changes in the Keystone build system. I'll put together a fix for that part. (It currently will not correctly copy the demo binaries into the keystone overlay directory to be included in the image, this can be done manually)

I will try to clean up some of the script to prevent this from happening if you run multiple times, or with a bad environment.

**Comment by dkohlbre** [comment]
See the changes pushed in https://github.com/keystone-enclave/keystone-demo/issues/10
They make it a bit more clear how to run the quick setup scripts.
My suggest is to (in keystone-demo) run:
``` 
rm -rf libsodium_builds
./quick-setup.sh
```

**Comment by hadaroren11** [comment]
i removed keystone-demo entirely and cloned it again and did everything from the top.
i ran the new quickstart script and it passed successfully.
later i ran manually the make copyboys as it says on docs/running_and_buliding and also everything was great.
when i tries to run get attestation I got again :
![image](https://user-images.githubusercontent.com/57223550/71247816-2f70c000-2322-11ea-93ff-4bd46fcad783.png)

**Comment by dkohlbre** [comment]
Please be sure to follow all directions the script gives. The binaries we not copied into the image.

**Comment by hadaroren11** [comment]
so is the binaries are not copied to the image, so the get attestation script will not work, because he need those files.
i ran the quick start script only once and it passed successfully. 
then i did make in keystone and then the get attestation script.
i really don’t know what is wrong...

**Comment by dkohlbre** [comment]
I suspect you need to rebuild the image to include the binaries. The script says to do this after it runs correctly: `make image` in the Keystone build dir. I tested this in a clean build of everything and its still functional.

Unfortunately, the build system, modules, etc in Keystone are changing right now, so the documentation in the demo isn't 100% up-to-date. When it stabilizes we'll update the docs here.
